dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -368,7 +370,17 @@ protected void exported() {
                 mapServiceName(url, serviceNameMapping, scheduledExecutor);
             }
         });
+
         onExported();
+
+        if (hasRegistrySpecified()) {
+            getScopeModel().getDeployer().getApplicationDeployer().exportMetadataService();
```

Comment:
Depends on MetadataService here might not a good design

Reply:
Consider provide a solution for this issue with https://github.com/apache/dubbo/pull/12750

Reply:
It seems that the full service export process can move from ServiceConfig to a new ModuleLifecycle implement? This might solve this problem

Reply:
Yes, I think so, this might need some extra efforts to merge this change with Lifecycle mechanism.